### PR TITLE
8338018: Rename ClassPrelinker to AOTConstantPoolResolver

### PR DESCRIPTION
**Pull request by iklam**
This is the 2nd PR for [JEP 483: Ahead-of-Time Class Loading & Linking](https://bugs.openjdk.org/browse/JDK-8315737).

A simple renaming of the `ClassPrelinker` class to `AOTConstantPoolLinker`, so that the name is consistent with new classes that will be introduced in subsequent PRs for JEP 483 (`AOTClassLinker`, `AOTLinkedClassTable`, and `AOTLinkedClassBulkLoader`).

-----
See [here](https://bugs.openjdk.org/browse/JDK-8315737) for the sequence of dependent RFEs for implementing JEP 483.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Integration blocker
&nbsp;⚠️ Dependency #20516 must be integrated first

### Issue
 * [JDK-8338018](https://bugs.openjdk.org/browse/JDK-8338018): Rename ClassPrelinker to AOTConstantPoolResolver (**Enhancement** - P3)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) Review applies to [fed4dfed](https://git.openjdk.org/jdk/pull/20517/files/fed4dfedacbe948e86ef202f0e1f2efcc1634665)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20517/head:pull/20517` \
`$ git checkout pull/20517`

Update a local copy of the PR: \
`$ git checkout pull/20517` \
`$ git pull https://git.openjdk.org/jdk.git pull/20517/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20517`

View PR using the GUI difftool: \
`$ git pr show -t 20517`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20517.diff">https://git.openjdk.org/jdk/pull/20517.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20517#issuecomment-2316687481)

### CODE REVIEW

File: src/hotspot/share/cds/aotConstantPoolResolver.hpp [hunk 1]
```diff
@@ -39,7 +39,9 @@ class constantPoolHandle;
 class InstanceKlass;
 class Klass;
 
-// ClassPrelinker is used to perform ahead-of-time linking of ConstantPool entries
+template <typename T> class GrowableArray;
```
Comment (dholmes-ora) [inline src/hotspot/share/cds/aotConstantPoolResolver.hpp:42]: This doesn't seem part of the rename.

**Review by dholmes-ora — COMMENTED** [review]
Rename looks fine, but there seems to be an unrelated change.

Thanks

File: src/hotspot/share/cds/aotConstantPoolResolver.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/cds/aotConstantPoolResolver.hpp:42]: This header was using GrowableArray without declaring it. The problem is discovered after this header is moved earlier due to alphabetical sorting.

Comment (dholmes-ora) [reply src/hotspot/share/cds/aotConstantPoolResolver.hpp:42]: Okay - thanks for explaining.

**Review by dholmes-ora — APPROVED** [review]

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by dholmes-ora — APPROVED** [review]
Still good

**Comment by iklam** [comment]
Closing this PR as its changes have been combined into https://github.com/openjdk/jdk/pull/21642.

**Pull request closed without merging**